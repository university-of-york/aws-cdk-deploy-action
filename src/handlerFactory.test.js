const runCdkCommands = require('./cdkCommands');
const handlerFactory = require('./handlerFactory');
const core = require('@actions/core');
jest.mock('@actions/core', () => {
    const { setFailed } = jest.requireActual('@actions/core');

    return {
        setFailed,
        getInput: jest.fn(),
    };
});

jest.mock('./cdkCommands');

describe('handlerFactory', () => {
    beforeEach(() => jest.clearAllMocks());
    afterAll(() => jest.restoreAllMocks());

    it('should call run cdk commands with correct arguments', async () => {
        const userInput = {
            AWS_ROLE_NAME: 'role',
            AWS_STACK_NAME: 'stack',
            INFRASTRUCTURE_PATH: '.',
            AWS_REGION: 'some_region',
            SKIP_BOOTSTRAP: 'false',
        };

        core.getInput = jest.fn((value) => userInput[value]);

        // secrets
        process.env.AWS_ACCOUNT_ID = '1234';
        process.env.AWS_ACCESS_KEY_ID = 'secret_id';
        process.env.AWS_SECRET_ACCESS_KEY = 'secret_password';

        await handlerFactory();

        expect(runCdkCommands).toHaveBeenCalledTimes(1);
        expect(runCdkCommands.mock.calls[0][0]).toMatchSnapshot();
    });

    it('should call run cdk commands with user-defined arguments', async () => {
        const userInput = {
            AWS_ROLE_NAME: 'role',
            AWS_STACK_NAME: 'stack',
            INFRASTRUCTURE_PATH: '.',
            AWS_REGION: 'some_region',
            SKIP_BOOTSTRAP: 'false',
            CUSTOM_BOOTSTRAP_ARGUMENTS: '--testargument=abc,--all',
            CUSTOM_DEPLOY_ARGUMENTS: '--testargument=abc,--all',
        };

        core.getInput = jest.fn((value) => userInput[value]);

        // secrets
        process.env.AWS_ACCOUNT_ID = '1234';
        process.env.AWS_ACCESS_KEY_ID = 'secret_id';
        process.env.AWS_SECRET_ACCESS_KEY = 'secret_password';

        await handlerFactory();

        expect(runCdkCommands).toHaveBeenCalledTimes(1);
        expect(runCdkCommands.mock.calls[0][0]).toMatchSnapshot();
    });
});
