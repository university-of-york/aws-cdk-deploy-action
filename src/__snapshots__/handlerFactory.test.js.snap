// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handlerFactory should call run cdk commands with correct arguments 1`] = `
Object {
  "AWS_ACCESS_KEY_ID": "secret_id",
  "AWS_ACCOUNT_ID": "1234",
  "AWS_REGION": "some_region",
  "AWS_ROLE_NAME": "role",
  "AWS_SECRET_ACCESS_KEY": "secret_password",
  "AWS_STACK_NAME": "stack",
  "CUSTOM_BOOTSTRAP_ARGUMENTS": undefined,
  "CUSTOM_DEPLOY_ARGUMENTS": undefined,
  "INFRASTRUCTURE_PATH": ".",
  "SKIP_BOOTSTRAP": "false",
}
`;

exports[`handlerFactory should call run cdk commands with user-defined arguments 1`] = `
Object {
  "AWS_ACCESS_KEY_ID": "secret_id",
  "AWS_ACCOUNT_ID": "1234",
  "AWS_REGION": "some_region",
  "AWS_ROLE_NAME": "role",
  "AWS_SECRET_ACCESS_KEY": "secret_password",
  "AWS_STACK_NAME": "stack",
  "CUSTOM_BOOTSTRAP_ARGUMENTS": "--testargument=abc,--all",
  "CUSTOM_DEPLOY_ARGUMENTS": "--testargument=abc,--all",
  "INFRASTRUCTURE_PATH": ".",
  "SKIP_BOOTSTRAP": "false",
}
`;
